utils/s2n_fork_detection.c

```diff
@@ -153,10 +153,16 @@ static void s2n_pthread_atfork_on_fork(void)
 
 static S2N_RESULT s2n_inititalise_pthread_atfork(void)
 {
-    /* Register the fork handler pthread_atfork_on_fork that is excuted in the
+    /* Register the fork handler pthread_atfork_on_fork that is executed in the
      * child process after a fork.
+     *
+     * The handler is not installed on OpenBSD due to an OpenBSD libc bug which
+     * is triggered by multi-generational forking of multi-threaded processes.
+     * The grandchild process will deadlock when trying to fork a great-grandchild.
```

Comment:
Please link bug report.

Reply:
Moved the comment to the new function `s2n_is_pthread_atfork_supported()` and added a link to the bug report there.